### PR TITLE
Improve performance of /points

### PR DESCRIPTION
**Pull request by def-**
Worst case seems to be same performance, best case 3 ms instead of 3 s previously

### CODE REVIEW

File: src/game/server/score.cpp [hunk 1]
```diff
@@ -1045,12 +1045,14 @@ bool CScore::ShowPointsThread(IDbConnection *pSqlServer, const ISqlData *pGameDa
 			"FROM ("
 				"SELECT RANK() OVER w AS Rank, Points, Name "
 				"FROM %s_points "
+				"WHERE Points >= (select Points from record_points where Name = ?) "
```
Comment (heinrich5991) [inline src/game/server/score.cpp:1048]: `record_points` shouldn't be spelled out.

**Comment by heinrich5991** [comment]
I think
```sql
SELECT 
    (SELECT COUNT(name) FROM record_points WHERE points > (SELECT points FROM record_points WHERE name = 'heinrich5991')),
    (SELECT points FROM record_points WHERE name = 'heinrich5991');
```
would be even faster.

**Comment by def-** [comment]
Ah right, no need for the fancy rank at all.

**Comment by Zwelf** [comment]
Good find. Can't test the performance difference by myself yet, since I didn't import points. But the query looks good and is still working :+1:.
bors r+

**Comment by BannZay** [comment]
Is there index on points  from record_points ?
It seems like points field is a good candidate to become a clustered one tho

**Comment by def-** [comment]
I think an index doesn't really help when the rank() calculation is what's expensive.
```
MariaDB [(none)]> show index from teeworlds.record_points;
+---------------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| Table         | Non_unique | Key_name | Seq_in_index | Column_name | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment |
+---------------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| record_points |          0 | Name     |            1 | Name        | A         |      237468 |     NULL | NULL   |      | BTREE      |         |               |
| record_points |          1 | points   |            1 | Points      | A         |        6249 |     NULL | NULL   | YES  | BTREE      |         |               |
+---------------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
2 rows in set (0.000 sec)
```

**Pull request merged**